modules/misc/nixpkgs.nix

```diff
@@ -0,0 +1,29 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  nixpkgsModule = import <nixpkgs/nixos/modules/misc/nixpkgs.nix> {
+    inherit lib pkgs;
+    config.nixpkgs = removeAttrs config.nixpkgs [ "enable" ];
+  };
+
+in
+
+{
+  # We reuse the options from the NixOS module but we remove the
+  # `system` attribute since it is not particularly relevant for a
+  # user environment. We also add an `enable` option to allow the user
+  # to decide whether Home Manager should manage the Nixpkgs
+  # configuration.
+  options = {
+    nixpkgs =
+      removeAttrs nixpkgsModule.options.nixpkgs [ "system" ]
+      // {
+        enable = mkEnableOption "management of Nixpkgs configuration";
```

Comment:
I think it would be better to always enable it, but have it combine the default overlays with the ones defined in `nixpkgs.overlays` (which is by default `[]`), the same for `nixpkgs.config` (default `{}`). While this would complicate it quite a bit, since the [config and overlays defaults](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/impure.nix#L27-L73) would need to be extracted (this should really be a proper lib function anyways), I think it's the much more correct and expected behavior. I'm not sure how to do that properly though

Reply:
There's little cost to always enabling it, right? If the module is enabled but the option is not set, then we will fall back to the other options as we would have otherwise.

Reply:
@michaelpj The defaults for `config` and `overlays` in the nixpkgs module are `{}` and `[]`, respectively. So if the nixpkgs module is unconditionally enabled then it would be a breaking change for anybody who, e.g., use `home.packages` to install a package defined in an overlay inside `~/.config/nixpkgs/overlays`.

Reply:
I think it's worth copying the nixpkgs module from nixos and modifying it to our needs: extend the types with `nullOr`, and defaulting to null, in which case the option isn't used.

Reply:
Fair enough. I was hoping to avoid copying it over but using `nullOr` does make things a bit smoother. Force pushed a new version that does this. Please have a look.